app/(dashboard)/manage-repos/actions.ts

```diff
@@ -0,0 +1,18 @@
+"use server";
+
+import { activateRepository, deactivateRepository, fetchRepoDetails } from "@/lib/github/services/repository";
+
+// Activates a single repository
+export const activateRepoAction = async (id: string, userId: string) => {
```

Comment:
SECURITY RISK:
server actions are not secure (no authentication or authorization). They are public endpoints meaning everyone can perform this action if they know the id and the userId of an authorized user.
`userId` should be get from the current session and session and authorization should be checked in every server action.